internal/service/datazone/project.go

```diff
@@ -0,0 +1,452 @@
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+
+			"glossary_terms": schema.ListAttribute{
+				CustomType:  fwtypes.ListOfStringType,
+				ElementType: types.StringType,
+
+				Validators: []validator.List{
+					listvalidator.SizeBetween(1, 20),
+					listvalidator.ValueStringsAre(stringvalidator.RegexMatches(regexache.MustCompile(`^[a-zA-Z0-9_-]{1,36}$`), "must conform to: ^[a-zA-Z0-9_-]{1,36}$ ")),
+				},
+				Optional: true,
+			},
+
+			names.AttrName: schema.StringAttribute{
+				Validators: []validator.String{
+					stringvalidator.RegexMatches(regexache.MustCompile(`^[\w -]+$`), "must conform to: ^[\\w -]+$ "),
+					stringvalidator.LengthBetween(1, 64),
+				},
+				Required: true,
+			},
+			"created_by": schema.StringAttribute{
+				Computed: true,
+			},
+			names.AttrID: framework.IDAttribute(),
+
+			names.AttrCreatedAt: schema.StringAttribute{
+				CustomType: timetypes.RFC3339Type{},
+				Computed:   true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+
+			"failure_reasons": schema.ListAttribute{
+				CustomType: fwtypes.NewListNestedObjectTypeOf[dsProjectDeletionError](ctx),
+				Computed:   true,
+			},
+
+			"last_updated_at": schema.StringAttribute{
+				CustomType: timetypes.RFC3339Type{},
+				Computed:   true,
+			},
+			"project_status": schema.StringAttribute{
+				CustomType: fwtypes.StringEnumType[awstypes.ProjectStatus](),
+				Computed:   true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"skip_deletion_check": schema.BoolAttribute{
+				Optional: true,
+				PlanModifiers: []planmodifier.Bool{
+					boolplanmodifier.UseStateForUnknown(),
+				},
+			},
+		},
+		Blocks: map[string]schema.Block{
+			names.AttrTimeouts: timeouts.Block(ctx, timeouts.Opts{
+				Create: true,
+				Delete: true,
+			}),
+		},
+	}
+}
+
+func (r *resourceProject) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan resourceProjectData
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	conn := r.Meta().DataZoneClient(ctx)
+
+	var validateDomain datazone.GetDomainInput
+	validateDomain.Identifier = plan.DomainId.ValueStringPointer()
+	_, err := conn.GetDomain(ctx, &validateDomain)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			create.ProblemStandardMessage(names.DataZone, create.ErrActionCreating, ResNameProject, plan.Name.String(), err),
+			err.Error(),
+		)
+		return
+	}
+
+	in := &datazone.CreateProjectInput{
+		Name:             aws.String(plan.Name.ValueString()),
+		DomainIdentifier: aws.String(plan.DomainId.ValueString()),
+	}
+	if !plan.Description.IsNull() {
+		in.Description = aws.String(plan.Description.ValueString())
+	}
+	if !plan.GlossaryTerms.IsNull() {
+		in.GlossaryTerms = aws.ToStringSlice(flex.ExpandFrameworkStringList(ctx, plan.GlossaryTerms))
+	}
```

Comment:
We should use `flex.Expand` here. AutoFlex should work in this situation, which will reduce some lines of code